compiler/generator_rust.c

```diff
@@ -0,0 +1,1347 @@
+    g->V[0] = p->name;
+    writef(g, "~M~V0 = true;~N", p);
+}
+
+static void generate_unset(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->V[0] = p->name;
+    writef(g, "~M~V0 = false;~N", p);
+}
+
+static void generate_fail(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    generate(g, p->left);
+    if (!g->unreachable) write_failure(g);
+}
+
+/* generate_test() also implements 'reverse' */
+
+static void generate_test(struct generator * g, struct node * p) {
+
+    struct str * savevar = vars_newname(g);
+    int keep_c = K_needed(g, p->left);
+
+    write_comment(g, p);
+
+    if (keep_c) {
+        write_savecursor(g, p, savevar);
+    }
+
+    generate(g, p->left);
+
+    if (!g->unreachable) {
+        if (keep_c) {
+            write_restorecursor(g, p, savevar);
+        }
+    }
+    str_delete(savevar);
+}
+
+static void generate_do(struct generator * g, struct node * p) {
+
+    struct str * savevar = vars_newname(g);
+    int keep_c = K_needed(g, p->left);
+    write_comment(g, p);
+    if (keep_c) write_savecursor(g, p, savevar);
+    g->failure_label = new_label(g);
+    int label = g->failure_label;
+    str_clear(g->failure_str);
+
+    wsetlab_begin(g, label);
+    generate(g, p->left);
+    wsetlab_end(g, label);
+    g->unreachable = false;
+
+    if (keep_c) write_restorecursor(g, p, savevar);
+    str_delete(savevar);
+}
+
+static void generate_GO(struct generator * g, struct node * p, int style) {
+
+    int end_unreachable = false;
+    struct str * savevar = vars_newname(g);
+    int keep_c = style == 1 || repeat_restore(g, p->left);
+
+    int a0 = g->failure_label;
+    struct str * a1 = str_copy(g->failure_str);
+
+    int golab = new_label(g);
+    g->I[0] = golab;
+    write_comment(g, p);
+    w(g, "~M'golab~I0: loop{~N~+");
+    if (keep_c) write_savecursor(g, p, savevar);
+
+    g->failure_label = new_label(g);
+    wsetlab_begin(g, g->failure_label);
+    generate(g, p->left);
+
+    if (g->unreachable) {
+        /* Cannot break out of this loop: therefore the code after the
+         * end of the loop is unreachable.*/
+        end_unreachable = true;
+    } else {
+        /* include for goto; omit for gopast */
+        if (style == 1) write_restorecursor(g, p, savevar);
+        g->I[0] = golab;
+        w(g, "~Mbreak 'golab~I0;~N");
+    }
+    g->unreachable = false;
+    w(g, "~-~M}~N");
+    if (keep_c) write_restorecursor(g, p, savevar);
+
+    g->failure_label = a0;
+    str_delete(g->failure_str);
+    g->failure_str = a1;
+
+    write_check_limit(g, p);
+    write_inc_cursor(g, p);    
+    write_block_end(g);
+
+    str_delete(savevar);
+    g->unreachable = end_unreachable;
+}
+
+static void generate_loop(struct generator * g, struct node * p) {
+
+    struct str * loopvar = vars_newname(g);
+    write_comment(g, p);
+    w(g, "~Mfor _ in 0..");
+    generate_AE(g, p->AE);
+    writef(g, "~N", p);
+    writef(g, "~M{", p);
```

Comment:
It is more consistent with the rest of the generated code for the `{` to be on the same line as the `for`.

Also, the `{` should have a new line after it.